zrml/prediction-markets/src/lib.rs

```diff
@@ -474,26 +472,71 @@ mod pallet {
                     T::SimpleDisputes::on_dispute(&disputes, &market_id, &market)?
                 }
             }
-            Self::remove_last_dispute_from_market_ids_per_dispute_block(&disputes, &market_id)?;
+
             Self::set_market_as_disputed(&market, &market_id)?;
             let market_dispute = MarketDispute { at: curr_block_num, by: who, outcome };
             <Disputes<T>>::try_mutate(market_id, |disputes| {
                 disputes.try_push(market_dispute.clone()).map_err(|_| <Error<T>>::StorageOverflow)
             })?;
-            // each dispute resets dispute_duration
-            let dispute_duration_ends_at_block =
-                curr_block_num.saturating_add(market.deadlines.dispute_duration);
-            <MarketIdsPerDisputeBlock<T>>::try_mutate(dispute_duration_ends_at_block, |ids| {
-                ids.try_push(market_id).map_err(|_| <Error<T>>::StorageOverflow)
-            })?;
-
             Self::deposit_event(Event::MarketDisputed(
                 market_id,
                 MarketStatus::Disputed,
                 market_dispute,
             ));
-            // TODO(#782): add court benchmark
-            Ok((Some(T::WeightInfo::dispute_authorized(num_disputes, CacheSize::get()))).into())
+            Ok((Some(T::WeightInfo::dispute_authorized())).into())
+        }
+
+        /// Resolve the market,
+        /// if the dispute mechanism was unable to come to a conclusion in a specified time.
+        ///
+        /// # Weight
+        ///
+        /// Complexity: `O(n)`, where `n` is the number of outstanding disputes.
+        #[pallet::weight(
+            T::WeightInfo::resolve_expired_mdm_authorized_categorical()
+                .max(T::WeightInfo::resolve_expired_mdm_authorized_scalar())
+        )]
+        #[transactional]
+        pub fn resolve_failed_mdm(
+            origin: OriginFor<T>,
+            #[pallet::compact] market_id: MarketIdOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?;
+            let market = <zrml_market_commons::Pallet<T>>::market(&market_id)?;
+            ensure!(market.status == MarketStatus::Disputed, Error::<T>::InvalidMarketStatus);
+
+            let disputes = Disputes::<T>::get(market_id);
+
+            // TODO(#782): use multiple benchmarks paths for different dispute mechanisms
+            let _has_failed = match market.dispute_mechanism {
```

Comment:
The reason for this style is because clippy would complain if it is called `has_failed`, because in the runtime benchmark case the `let _has_failed = true;` would just overwrite the _has_failed of the MDMs. Without the benchmark case, I could just use `has_failed`.

Reply:
https://github.com/zeitgeistpm/zeitgeist/commit/ceddfa944e675656035e025b0e7ff3aed0e642fe